code/gateway/gatewaylib.q

```diff
@@ -92,3 +92,199 @@ getserverscross:{[req;att;besteffort]
+            where{(y in key x[`dataaccess;`tablename])&`dataaccess in key x}[;options`tablename]each attributes;
+    timecolumn:$[`timecolumn in key options;options`timecolumn;`time];
+    // get time segment based on timecolumn specified
+    modquery:update inftc:.[inftc;(::;1);:;.[inftc;(::;1;timecolumn)]]from modquery;
+    // union join based on serverid
+    querytable:0!(`serverid xkey update serverid:(first each key query)from value query)uj`serverid xkey modquery;
+    // convert times to timestamps
+    querytable:update starttime:`timestamp$starttime,endtime:(`timestamp$endtime)+?[-14h=type each endtime;0D23:59:59.999999999;0]from querytable;
+    // modify starttime and endtime based on stripe
+    querytable:update 
+        // if no overlap return empty list
+        timeoverlaps:{[st;et;tc] $[(et<tc 0)|st>tc 1;`timestamp$();($[st<tc 0;tc 0;st];$[et>tc 1;tc 1;et])]}'[starttime;endtime;inftc[;1]]
+            from querytable where serverid in modquery`serverid;
+    querytable:update starttime:timeoverlaps[;0],endtime:timeoverlaps[;1] from querytable where serverid in modquery`serverid; 
+    querytable:enlist[`timeoverlaps]_querytable;
+
+    // filter queries not required
+    querytable:$[i:`instruments in key options;
+        select from querytable where(0=count each inftc)|(0<count each inftc[;1])&(not null each starttime)&0<count each instruments;
+        select from querytable where(0=count each inftc)|(0<count each inftc[;1])&not null each starttime];
+    // if no results
+    if[0=count querytable;'`$"gateway error - no info found for that table name and time range."];
+    // convert serverid atoms into their respective serverid lists
+    querytable:update serverid:{x where{any x in y}[;y]each x}[options`procs;serverid],
+        // get servertype
+        servertype:`${string .gw.servers'[x]`servertype}serverid,
+        // convert procs into procname if striped
+        procs:.gw.servers[;`attributes;`procname]@/:serverid from querytable;
+    querytable:update procs:servertype from querytable where not(last each serverid)in modquery`serverid;
+
+    // check if instruments are contained in stripe mapping
+    segids:exec distinct[segid]except ` from querytable;
+    if[all[i,count segids];
+        if[0 < count missinginstruments[options`instruments];
+            // if a queried instrument is not in stripe mapping, reload stripe mapping to bring up-to-date mapping
+            .ds.getstripemapping[];
+            // create messaging if instrument is not in mapping
+            if[[missing:missinginstruments[options`instruments]; 0 < count missing];
+                .lg.o[`.gw.adjustqueriesstripe;raze("No intra-day data for the following instruments: ",(string missing))];
+                    // exit script for intra-day only queries
+                if[[(count options`instruments) = count missing;not `hdb in exec servertype from querytable];
```

Comment:
Yes you're right, it will execute dependant on the second cond which was not the intention.  This must've been one I had missed fixing

Reply:
Yes that's right - it would only execute dependant on the second condition which was not the intention.  I will implement an ampersand to ensure both return true to pass this conditional.